Automattic/wp-calypso
# Full Site Editing: Update the Content Slot block

#### Changes proposed in this Pull Request

* Rename the block from "Page Content" to **Content Slot**.
* Replace the temporary post selector approach (a simple `select` with the latest 10 posts) with an autosuggestion field roughly inspired by [`URLInput`](https://github.com/WordPress/gutenberg/tree/354c1ff6b5337837eb4183d4eb8302ffeccd1be8/packages/block-editor/src/components/url-input#urlinput).

Note: the accessibility interactions between the input field and the popover should be improved.
Core handles keyboard events to move the cursor from one side to the other.
I have omitted it because it's a pretty good chunk of code, but we should take care of it in a follow up.
Note: the render callback for this block depends on the theme, and will be tackled in a follow-up.

![Apr-23-2019 18-47-49](https://user-images.githubusercontent.com/2070010/56604673-745d1f00-65fa-11e9-9374-d65f4e5adecd.gif)

#### Testing instructions

* Follow the [README](https://github.com/Automattic/wp-calypso/blob/master/apps/full-site-editing/README.md) to build the plugin. (If the script fails, please run `npm distclean` and `npm ci` first (required because of a recent [`calypso-build`](https://github.com/Automattic/wp-calypso/pull/32295) change).
* In the block editor, try adding the Content Slot block.
* Type the title of any public entity (posts, pages, `public` CPTs), and make sure the results are as expected.
* Click on one of the suggestions: it should be displayed into the block.

Fixes #32492
